pwiz_tools/Skyline/SettingsUI/ViewLibraryDlg.cs

```diff
@@ -1384,7 +1384,10 @@ private void comboFilterCategory_SelectedIndexChanged(object sender, EventArgs e
                 FilterAndUpdate();
             }
 
-            textPeptide.Focus(); // Assume that the next thing the user wants to do is work with the filter value
+            if (!Program.FunctionalTest)

```

Comment:
Why is this necessary or better? We don't usually have code like this. It should at least be really well commented as to why it is there to avoid others writing their UI like this without justification.

Reply:
It's a @nickshulman suggestion.

Reply:
Oops. My suggestion should have probably been:
`if (!Program.SkylineOffscreen)`
We don't want Skyline stealing the focus when it's supposed to be running offscreen. But, it's fine to steal the focus when not running off screen.

I am not sure that it makes sense to have the "textPeptide.Focus()" call here at all. We normally never have to set the focus to controls on a form when they are first shown.

Reply:
The context here is that the user has just operated the "what kind of thing do I want to search for" control, it's just a convenience to swap focus to the "what thing do I want to search for". I was annoyed at the extra click, there's really no where else I'd want to click next.

Reply:
But yes, that makes sense to avoid doing when running offscreen.

Reply:
Oh, yes, I see. That Focus() makes sense.

I might be wrong about the "if (!Program.SkylineOffscreen)" being necessary. I'm not 100% sure that calling "Focus()" on a control actually takes the focus if the form is not already active...

Reply:
We certainly see that in the code, especially around graphs, so I expect it's probably a good thing, certainly harmless anyway.

Reply:
I'm pretty sure I'm 100% wrong about that being necessary.
I tried writing a test which keeps setting focus to text boxes, and it never seemed to steal the focus from whatever other app I was using.
The really important thing for preventing focus stealing is the `override bool ShowWithoutActivation` in the class "FormEx".

Reply:
Excellent, I'll back it out.